src/Microsoft.Crank.Agent/Controllers/JobsController.cs

```diff
@@ -367,15 +375,18 @@ public async Task<IActionResult> UploadAttachmentZip(int id)
 
             var tempFilename = Path.GetTempFileName() + ".zip";
 
-            await SaveBodyAsync(tempFilename);
-
-            job.LastDriverCommunicationUtc = DateTime.UtcNow;
-
             try
             {
+                job.LastDriverCommunicationUtc = DateTime.UtcNow;
+                await SaveBodyAsync(job, tempFilename);
+
```

Comment:
Why did you change the order of these two lines?

Reply:
Doesn't matter much, this will be updated every second by the loop which is more than the 10s timeout.